examples/plot_vgm_error.py

```diff
@@ -0,0 +1,256 @@
+    neff_singlerange = xdem.spatialstats.neff_circ(area, [(params1[0], 'Sph', params1[1])])
+
+    # For a double-range model
+    neff_doublerange = xdem.spatialstats.neff_circ(area, [(params2[0], 'Sph', params2[1]),
+                                                   (params2[2], 'Sph', params2[3])])
+
+    # Convert into a standard error
+    stderr_singlerange = np.nanstd(dh.data)/np.sqrt(neff_singlerange)
+    stderr_doublerange = np.nanstd(dh.data)/np.sqrt(neff_doublerange)
+    list_stderr_singlerange.append(stderr_singlerange)
+    list_stderr_doublerange.append(stderr_doublerange)
+
+# %%
+# We also compute an empirical error based on intensive Monte-Carlo sampling ("patches" method) over the data grid
+# (Dehecq et al. (2020), Hugonnet et al., in prep), which is integrated in :func:`xdem.spatialstats.patches_method`.
+# Here, we sample fewer areas to avoid for the patches method to run over long processing times. We increase
+# exponentially from areas of 5 pixels to areas of 10000 pixels.
+areas_emp = [10 * 400 * 2 ** i for i in range(10)]
+for area_emp in areas_emp:
+
+    #  First, sample intensively circular patches of a given area, and derive the mean elevation differences
+    df_patches = xdem.spatialstats.patches_method(dh.data.data, gsd=dh.res[0], area=area_emp, nmax=200)
+    # Second, estimate the dispersion of the means of each patch, i.e. the standard error of the mean
+    stderr_empirical = np.nanstd(df_patches['mean'].values)
+    list_stderr_empirical.append(stderr_empirical)
+
+fig, ax = plt.subplots()
+plt.plot(np.asarray(areas)/1000000, list_stderr_singlerange, label='Single-range spherical model')
+plt.plot(np.asarray(areas)/1000000, list_stderr_doublerange, label='Double-range spherical model')
+plt.scatter(np.asarray(areas_emp)/1000000, list_stderr_empirical, label='Empirical estimate', color='black', marker='x')
+plt.xlabel('Averaging area (km²)')
+plt.ylabel('Uncertainty in the mean elevation difference (m)')
+plt.xscale('log')
+plt.yscale('log')
+plt.legend()
+
+# %%
+# We demonstrate that using a single-range variogram highly underestimates the measurement error integrated over an area,
+# by over a factor of ~100 for large surface areas. Using a double-range variogram brings us closer to the empirical error.
+# **But, in this case, the error is still too small. Why?**
+# The small size of the sampling area against the very large range of the noise means that we might no verify the
+# assumption of second-order stationarity (see :ref:`spatialstats`). We might be missing even longer range correlations
+# in our analysis, due to the limits of the variogram sampling.
+# In other words, a small part of the variance could be fully correlated over a large part of the grid, i.e. have a
+# vertical bias.
+# As a first guess for this, let's examine the difference between mean and median to gain some insight on the central
+# tendency of our sample:
+
+diff_med_mean = np.nanmean(dh.data.data)-np.nanmedian(dh.data.data)
```

Comment:
Why are you using dh.data.data? This means you're ignoring the mask (so you're including glaciers too).
This made me just realize that the test data set has NaNs ! This should be fixed, and it should be set to nodata instead. Added it as issue #196.

Reply:
Without the NaNs you could do `dh.data.mean() - np.ma.median(dh.data)`.

Reply:
Yes it has some!
```
np.count_nonzero(np.isnan(dh.data))
Out[14]: 4630
```